### PR TITLE
Updated aes-calculated.r for pairlists

### PR DESCRIPTION
**Pull request by NikNakk**
Updated to correct code to handle formals of anonymous function and zero-length names. This fixes the issue just posted at https://github.com/hadley/ggplot2/issues/1154 and discussed on Stack Overflow at http://stackoverflow.com/questions/31212054/functions-inside-aes .

### CODE REVIEW

File: R/aes-calculated.r [hunk 1]
```diff
@@ -28,13 +28,19 @@ strip_dots <- function(expr) {
   if (is.atomic(expr)) {
     expr
   } else if (is.name(expr)) {
-    as.name(gsub(match_calculated_aes, "\\1", as.character(expr)))
+    expr_ch <- as.character(expr)
+    if (nchar(expr_ch) > 0) {
+      as.name(gsub(match_calculated_aes, "\\1", as.character(expr)))
+    }
```
Comment (hadley) [inline R/aes-calculated.r]: Can you please match style?

**Comment by hadley** [comment]
Thanks. Can you please add a unit test or two and a bullet to NEWS?

**Comment by NikNakk** [comment]
I've added a unit test for a simple anonymous function to test-layer.r and a line to the bottom of the current NEWS. I've also shortened the line length of one of my updated lines by removing a redundant `as.character`. Was this the style issue or something else? I've looked through your style guide and can't find any other issues. In terms of line length, test-layer.r already has quite a few lines >60 characters and so I've not tried to shorten my lines to that length. Hope that's ok.

File: R/aes-calculated.r [hunk 2]
```diff
@@ -28,13 +28,19 @@ strip_dots <- function(expr) {
   if (is.atomic(expr)) {
     expr
   } else if (is.name(expr)) {
-    as.name(gsub(match_calculated_aes, "\\1", as.character(expr)))
+    expr_ch <- as.character(expr)
+    if (nchar(expr_ch) > 0) {
+      as.name(gsub(match_calculated_aes, "\\1", expr_ch))
+    }
```
Comment (hadley) [inline R/aes-calculated.r]: This is the style issue - `}` should be on the same line as `else`

**Comment by NikNakk** [comment]
Sorry, normally put `} else {`! Somehow managed to overlook that.

**Comment by hadley** [comment]
Thanks!

**Pull request merged**